File: qc-storage/src/main/java/com/eager/questioncloud/storage/creator/CreatorStatisticsEntity.java
package com.eager.questioncloud.storage.creator;

import com.eager.questioncloud.core.domain.creator.model.CreatorStatistics;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@Table(name = "creator_statistics")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class CreatorStatisticsEntity {
    @Id
    private Long creatorId;

    @Column
    private int subscribeCount;

    @Column
    private int salesCount;

    @Column
    private Double averageRateOfReview;

    @Builder
    public CreatorStatisticsEntity(Long creatorId, int subscribeCount, int salesCount, Double averageRateOfReview) {
        this.creatorId = creatorId;
        this.subscribeCount = subscribeCount;
        this.salesCount = salesCount;
        this.averageRateOfReview = averageRateOfReview;
    }

    public static CreatorStatisticsEntity from(CreatorStatistics creatorStatistics) {
        return CreatorStatisticsEntity.builder()
            .creatorId(creatorStatistics.getCreatorId())
            .subscribeCount(creatorStatistics.getSubscribeCount())
            .salesCount(creatorStatistics.getSaleCount())
            .averageRateOfReview(creatorStatistics.getAverageRateOfReview())
            .build();
    }
}

File: qc-core/src/main/kotlin/com/eager/questioncloud/core/domain/creator/infrastructure/entity/CreatorStatisticsEntity.kt
package com.eager.questioncloud.core.domain.creator.infrastructure.entity

import com.eager.questioncloud.core.common.BaseCustomIdEntity
import com.eager.questioncloud.core.domain.creator.model.CreatorStatistics
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "creator_statistics")
class CreatorStatisticsEntity(
    @Id var creatorId: Long,
    @Column var salesCount: Int,
    @Column var reviewCount: Int,
    @Column var totalReviewRate: Int,
    @Column var averageRateOfReview: Double,
    isNewEntity: Boolean
) : BaseCustomIdEntity<Long>(isNewEntity) {
    fun toModel(): CreatorStatistics {
        return CreatorStatistics(
            creatorId,
            salesCount,
            reviewCount,
            totalReviewRate,
            averageRateOfReview
        )
    }

    companion object {
        fun createNewEntity(creatorStatistics: CreatorStatistics): CreatorStatisticsEntity {
            return CreatorStatisticsEntity(
                creatorStatistics.creatorId,
                creatorStatistics.salesCount,
                creatorStatistics.reviewCount,
                creatorStatistics.totalReviewRate,
                creatorStatistics.averageRateOfReview,
                true
            )
        }

        fun fromExisting(creatorStatistics: CreatorStatistics): CreatorStatisticsEntity {
            return CreatorStatisticsEntity(
                creatorStatistics.creatorId,
                creatorStatistics.salesCount,
                creatorStatistics.reviewCount,
                creatorStatistics.totalReviewRate,
                creatorStatistics.averageRateOfReview,
                false
            )
        }
    }

    override fun getId(): Long {
        return creatorId
    }
}
